packages/sdk/src/oracle/optimistic/types/state.ts

```diff
@@ -0,0 +1,158 @@
+import type Multicall2 from "../../../multicall2";
+import {
+  RequestState,
+  RequestKey,
+  Request as RequestByEvent,
+  RequestPrice,
+  ProposePrice,
+  DisputePrice,
+  Settle,
+} from "../../../clients/optimisticOracle";
+
+import { JsonRpcSigner, BigNumber, Web3Provider, FallbackProvider } from "../../common/types/ethers";
+import type { erc20, optimisticOracle, sortedRequests } from "../../common/services";
+import { Context, Memory } from "../../common/types/statemachine";
+
+// create partial picker: https://stackoverflow.com/questions/43159887/make-a-single-property-optional-in-typescript
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+export { Context, Memory };
+
+export type ChainServices = {
```

Comment:
this file is not new, its just copied almost exactly from previous state file. lots of duplicate data between this and common state file which i'll try to fix when i add the skinny OO client